UnityProject/Assets/Scripts/MapSaver/MapSaver.cs

```diff
@@ -403,31 +426,79 @@ public static string VectorIntToGitFriendlyPosition(Vector3Int pos)
 
 		public static string TileToString(LayerTile layerTile)
 		{
-			return layerTile.name + LayerChar + (int) layerTile.TileType;
+			return layerTile.name;
+		}
+
+
+		public static Matrix4x4 StringToMatrix4X4(string Stringy)
```

Comment:
```cs
public static Matrix4x4 StringToMatrix4x4(string Stringy)
{
    var Entries = Stringy.Split(",");
    var matrix = new Matrix4x4();

    for (int i = 0; i < 16; i++)
    {
        matrix[i / 4, i % 4] = float.Parse(Entries[i]);
    }

    return matrix;
}

```

note, i don't know if this is how it actually works; but use this as an example for not hardcoding stuff